core/bin/external_node/src/config/mod.rs

```diff
@@ -874,6 +1201,65 @@ impl ExternalNodeConfig<()> {
         })
     }
 
+    pub fn from_files(
+        general_config_path: PathBuf,
+        external_node_config_patch: PathBuf,
+        secrets_configs: PathBuf,
+        consensus: Option<PathBuf>,
+    ) -> anyhow::Result<Self> {
+        let general_config = read_yaml_repr::<proto::general::GeneralConfig>(general_config_path)
+            .context("failed decoding general YAML config")?;
+        let external_node_config =
+            read_yaml_repr::<proto::en::ExternalNode>(external_node_config_patch)
+                .context("failed decoding external node YAML config")?;
+        let secrets_config = read_yaml_repr::<proto::secrets::Secrets>(secrets_configs)
+            .context("failed decoding secrets YAML config")?;
+
+        let consensus = consensus
+            .map(read_yaml_repr::<proto::consensus::Config>)
+            .transpose()
```

Comment:
Why are these separate files required? I guess for secrets the reasoning is that we want to store secrets in a K8s secret (which is a moderately bogus reasoning because a node is not necessarily run on K8s), but what about other two? Why can't they be parsed from sections in the general configuration file?

Reply:
Once consensus is fully ready we move the genesis part of external config there and the rest to the general. 
I think for this transition period we could tolerate it

Reply:
But why wait to move it there? I think it should be possible and relatively easy to parse configs from a certain object path in a YAML document, and it would result in better UX at once.

Reply:
they have completely different semantic, don't want to mess them up